pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -1017,15 +1017,19 @@ public class Foo {
           language="java"
           since="1.0"
           message="Switch statements should have a default label"
+          typeResolution="true"
           class="net.sourceforge.pmd.lang.rule.XPathRule"
           externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#switchstmtsshouldhavedefault">
         <description>
 All switch statements should include a default option to catch any unspecified values.
         </description>
         <priority>3</priority>
         <properties>
+            <property name="version" value="2.0"/>
```

Comment:
just out of curiosity, any reason to move this rule to saxon?

at some point we will probably have to address the saxon vs jaxen performance... I think I did some tests a long time ago, and saxon was significantly slower, but it could be down mostly to Jaxen XPath rules actively attempting to use the rulechain, while saxon doesn't...